// @notifi-network/notifi-node/dist/client/NotifiClient.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const mutations_1 = require("../mutations");
const broadcastMessageImpl_1 = __importDefault(require("../mutations/broadcastMessageImpl"));
const queries_1 = require("../queries");
const types_1 = require("../types");
class NotifiClient {
    constructor(a) {
        this.a = a;
        this.logIn = async ({ sid, secret }) => {
            const input = { sid, secret };
            return await (0, mutations_1.logInFromServiceImpl)(this.a, { input });
        };
        this.sendSimpleHealthThreshold = async (jwt, _a) => {
            var { key, walletPublicKey, walletBlockchain } = _a, payload = __rest(_a, ["key", "walletPublicKey", "walletBlockchain"]);
            const message = (0, types_1.newSimpleHealthThresholdMessage)(payload);
            const input = {
                walletPublicKey,
                walletBlockchain,
                messageKey: key,
                messageType: message.type,
                message: JSON.stringify(message.payload),
            };
            const result = await (0, mutations_1.sendMessageImpl)(this.a, jwt, { input });
            if (!result) {
                throw new Error('Send message failed');
            }
        };
        this.sendBroadcastMessage = async (jwt, params) => {
            const result = await (0, broadcastMessageImpl_1.default)(this.a, jwt, params);
            if (result.id === null) {
                throw new Error('broadcast message failed');
            }
        };
        this.sendDirectPush = async (jwt, { key, walletPublicKey, walletBlockchain, message, template, type }) => {
            var _a, _b, _c;
            let directMessage;
            if (message !== undefined) {
                directMessage = (0, types_1.newDirectTenantMessage)({ message, type });
            }
            else if (template !== undefined) {
                directMessage = (0, types_1.newDirectTenantMessage)({
                    message: '',
                    type,
                    targetTemplates: {
                        SMS: (_a = template.smsTemplate) !== null && _a !== void 0 ? _a : undefined,
                        Email: (_b = template.emailTemplate) !== null && _b !== void 0 ? _b : undefined,
                        Telegram: (_c = template.telegramTemplate) !== null && _c !== void 0 ? _c : undefined,
                    },
                    templateVariables: template.variables,
                });
            }
            else {
                throw new Error('One of message or template must be set');
            }
            const input = {
                walletPublicKey,
                walletBlockchain,
                messageKey: key,
                messageType: directMessage.type,
                message: JSON.stringify(directMessage.payload),
            };
            const result = await (0, mutations_1.sendMessageImpl)(this.a, jwt, { input });
            if (!result) {
                throw new Error('Send message failed');
            }
        };
        this.deleteUserAlert = async (jwt, { alertId }) => {
            await (0, mutations_1.deleteUserAlertImpl)(this.a, jwt, { alertId });
            return alertId;
        };
        this.createTenantUser = async (jwt, { walletBlockchain, walletPublicKey }) => {
            const result = await (0, mutations_1.createTenantUserImpl)(this.a, jwt, {
                input: { walletBlockchain, walletPublicKey },
            });
            return result.id;
        };
        this.createDirectPushAlert = async (jwt, { userId, emailAddresses, phoneNumbers, telegramIds }) => {
            const input = {
                userId,
                emailAddresses: emailAddresses !== null && emailAddresses !== void 0 ? emailAddresses : [],
                phoneNumbers: phoneNumbers !== null && phoneNumbers !== void 0 ? phoneNumbers : [],
                telegramIds: telegramIds !== null && telegramIds !== void 0 ? telegramIds : [],
            };
            return await (0, mutations_1.createDirectPushAlertImpl)(this.a, jwt, {
                input,
            });
        };
        this.deleteDirectPushAlert = async (jwt, { alertId }) => {
            const input = { alertId };
            return await (0, mutations_1.deleteDirectPushAlertImpl)(this.a, jwt, { input });
        };
        this.getTenantConnectedWallet = async (jwt, params) => {
            return await (0, queries_1.getTenantConnectedWalletImpl)(this.a, jwt, params);
        };
    }
}
exports.default = NotifiClient;
