// @notifi-network/notifi-axios-utils/dist/utils/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NOTIFI_CONFIGS = exports.notifiConfigs = exports.makeRequest = exports.makeParameterLessRequest = exports.makeAuthenticatedRequest = exports.collectDependencies = void 0;
const axiosRequest_1 = require("./axiosRequest");
Object.defineProperty(exports, "makeAuthenticatedRequest", { enumerable: true, get: function () { return axiosRequest_1.makeAuthenticatedRequest; } });
Object.defineProperty(exports, "makeParameterLessRequest", { enumerable: true, get: function () { return axiosRequest_1.makeParameterLessRequest; } });
Object.defineProperty(exports, "makeRequest", { enumerable: true, get: function () { return axiosRequest_1.makeRequest; } });
const collectDependencies_1 = __importDefault(require("./collectDependencies"));
exports.collectDependencies = collectDependencies_1.default;
const notifiConfigs_1 = __importStar(require("./notifiConfigs"));
exports.notifiConfigs = notifiConfigs_1.default;
Object.defineProperty(exports, "NOTIFI_CONFIGS", { enumerable: true, get: function () { return notifiConfigs_1.NOTIFI_CONFIGS; } });
